### PR TITLE
Incorrect handling of arrays

### PR DESCRIPTION
**Pull request by abdulrahmancz**
Add square brackets for array items of stringified query string. Remove square brackets while parsing query string with array items. Updated tests and readme. Also related ti [issue#66](https://github.com/sindresorhus/query-string/issues/66).

### CODE REVIEW

**Comment by Aweary** [comment]
This is inconsistent with how the native `querystring` module works in Node.

```js
> querystring.stringify({ foo: ['bar', 'baz'] })
'foo=bar&foo=baz'
```

If the official specification doesn't clarify this behavior then it seems most reasonable to remain consistent with the platform.

**Comment by kevva** [comment]
This should definitely be behind a option like it is in https://github.com/ljharb/qs.

File: index.js [hunk 1]
```diff
@@ -80,11 +81,12 @@ exports.stringify = function (obj, opts) {
 				if (val2 === undefined) {
 					return;
 				}
+				var arrayKey = encode(key, encode) + '[]';
```
Comment (kevva) [inline index.js]: Add an empty line before this.

File: index.js [hunk 2]
```diff
@@ -36,7 +36,8 @@ exports.parse = function (str) {
 		var key = parts.shift();
 		var val = parts.length > 0 ? parts.join('=') : undefined;
 
-		key = decodeURIComponent(key);
+		key = decodeURIComponent(key)
+			.replace(/\[\]$/, '');
```
Comment (kevva) [inline index.js]: `.replace` doesn't need to be on its own line.

**Comment by sindresorhus** [comment]
Closing for lack of response.

**Pull request closed without merging**